### PR TITLE
Adding knowledge

### PR DESCRIPTION
**Pull request by khink**
Rebased the AndriesSHP "Adding_knowledge" branch on current master.

Merge conflicts were mainly due to imports having changed, some flake8 was fixed in the rebase.

### CODE REVIEW

File: gellish_communicator/Anything.py [hunk 1]
```diff
@@ -194,3 +214,43 @@ def __repr__(self):
         return("Idea: {} lh_uid: {} ({}) {} rh_uid: {}".
                format(self.uid, self.lh_obj.uid, self.rel_type.uid,
                       self.phrase_type_uid, self.rh_obj.uid))
+
+
+if __name__ == "__main__":
```
Comment (khink) [inline gellish_communicator/Anything.py]: Why is there a __main__ method here?

File: gellish_communicator/Create_output_file.py [hunk 2]
```diff
@@ -90,44 +90,51 @@ def Open_output_file(expressions, subject_name, lang_name, serialization):
...
+            file_writer.writerow(header3)
             for expression in expressions:
-                fileWriter.writerow(expression)
+                file_writer.writerow(expression)
 
             f.close()
 
             # Open the written file in a CSV viewer (e.g. Excel)
-            open_file(output_file)
+            # open_file(output_file)
```
Comment (khink) [inline gellish_communicator/Create_output_file.py]: If this was commented out, why not remove it altogether?

File: gellish_communicator/Anything.py (hunk 1 above)
Comment (AndriesSHP) [reply gellish_communicator/Anything.py]: The main method is/was meant for testing the module on its own.

File: gellish_communicator/Create_output_file.py (hunk 2 above)
Comment (AndriesSHP) [reply gellish_communicator/Create_output_file.py]: It is intended to activate it again after development and testing is completed.

**Pull request merged**